design/ceph/object/store.md

```diff
@@ -221,6 +221,275 @@ spec:
+                                                # This means that default poolPlacement is also mutually exclusive with spec.metadataPool and spec.dataPool
+        metadataPoolName: rgw-meta-pool         # [REQUIRED] results to rgw-meta-pool:<CephObjectStore.name>.index
+        dataPoolName: rgw-data-pool             # [REQUIRED] results to rgw-data-pool:<CephObjectStore.name>.data
+        dataNonECPoolName: ""                   # [OPTIONAL] results to rgw-data-pool:<CephObjectStore.name>.data.nonec
+        # [OPTIONAL] - Storage classes for default placement target.
+        # Any placement has default storage class named STANDARD pointing to dataPoolName.
+        # So stageClasses allows to define additional storage classes on top of STANDARD for given placement.
+        storageClasses:
+      # Pool placement target named "fast"
+      - name: "fast"
+        metadataPoolName: rgw-fast-meta-pool     # [REQUIRED] results to rgw-fast-meta-pool:<CephObjectStore.name>.fast.index
+        dataPoolName: rgw-fast-data-pool         # [REQUIRED] results to rgw-fast-data-pool:<CephObjectStore.name>.fast.data
+        dataNonECPoolName: ""                   # [OPTIONAL] results to rgw-fast-data-pool:<CephObjectStore.name>.fast.data.nonec
+        # [OPTIONAL] - Additional Storage classes for fast placement target. Provide alternatives to default STANDARD storage clsss.
+        # STANDARD storage class cannot be changed or removed and always points to placement dataPoolName to store object data.
+        storageClasses:
+      # Pool placement target named "cold"
+      - name: "cold"                            # [REQUIRED] - placement id. Can be arbitrary.
+        metadataPoolName: rgw-cold-meta-pool    # [REQUIRED] results to rgw-cold-meta-pool:<CephObjectStore.name>.cold.index
+        dataPoolName: rgw-cold-data-pool        # [REQUIRED] results to rgw-cold-data-pool:<CephObjectStore.name>.cold.data
+        dataNonECPoolName: ""                   # [OPTIONAL] results to rgw-cold-data-pool:<CephObjectStore.name>.cold.data.nonec
+        # [OPTIONAL] - Additional Storage classes for cold placement target. Provides alternatives to default STANDARD storage clsss.
+        # STANDARD storage class cannot be changed or removed and always points to placement dataPoolName to store object data.
+        storageClasses:
+          - name: GLACIER                       # [REQUIRED] StorageClass name. Can be arbitrary but it is better to use one of AWS x-amz-storage-class names.
+            dataPoolName: glacier-data-pool     # [REQUIRED] results to glacier-data-pool:<CephObjectStore.name>.glacier
+```
+
+Here we specified that pools `rgw-meta-pool` and `rgw-data-pool` will be used by default. We also added extra placements named `fast` and `cold` with corresponding data and metadata pools. We also introduced extra storage class `GLACIER` for `cold` placement, which can be used to override object data pool to `glacier-data-pool` within `cold` placement.
```

Comment:
I don't care about this for the design doc, but I do think that Anthony's points about `GLACIER` having special behavior in AWS that users might expect is probably apt.  

For our user-facing documentation, we might want to do as he suggested here and use `STANDARD-IA` or some other naming: https://github.com/rook/rook/pull/14455#discussion_r1692272310.

This may also be a slightly better doc for the S3 storage classes in final docs (I found it very helpful): https://aws.amazon.com/s3/storage-classes/

Reply:
sure, will use `STANDARD-IA`  in user docs